### PR TITLE
=clt #18232 Serializer for ClusterSingleton

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #18232

File: akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala [hunk 1]
```diff
@@ -111,6 +111,11 @@ final class ClusterSingletonManagerSettings(
     new ClusterSingletonManagerSettings(singletonName, role, removalMargin, handOverRetryInterval)
 }
 
+/**
+ * Marker trait for remote messages with special serializer.
+ */
+trait ClusterSingletonMessage extends Serializable
```
Comment (ktoso) [inline akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala]: can be `sealed`?

File: akka-cluster-tools/src/main/scala/akka/cluster/singleton/protobuf/ClusterSingletonMessageSerializer.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import akka.cluster.singleton.ClusterSingletonManager.Internal.HandOverInProgress
+import akka.cluster.singleton.ClusterSingletonManager.Internal.HandOverToMe
+import akka.cluster.singleton.ClusterSingletonManager.Internal.TakeOverFromMe
+import akka.serialization.BaseSerializer
+import akka.serialization.SerializationExtension
+import akka.serialization.SerializerWithStringManifest
+
+/**
+ * INTERNAL API: Serializer of ClusterSingleton messages.
+ * It is actually not using protobuf, but if we add more messages to
```
Comment (ktoso) [inline akka-cluster-tools/src/main/scala/akka/cluster/singleton/protobuf/ClusterSingletonMessageSerializer.scala]: sneaky, :+1:

**Comment by ktoso** [comment]
LGTM

File: akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala (hunk 1 above)
Comment (patriknw) [inline akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala]: yes, sealed it is

**Pull request merged**